### PR TITLE
recongnizing elementary OS and pretending it is Ubuntu Trusty

### PR DESCRIPTION
**Pull request by avidmych**
Elementary Freya is the latest release of Elementary OS: https://elementary.io/
It is based on Ubuntu 14.04 LTS, codename Trusty.

Tested it installs and works.
Works when running docker in sudo. Could not get it to work without sudo.

### CODE REVIEW

**Comment by calavera** [comment]
LGTM.

/cc @tianon, @jfrazelle

File: hack/install.sh [hunk 1]
```diff
@@ -120,6 +120,10 @@ do_install() {
 	if command_exists lsb_release; then
 		lsb_dist="$(lsb_release -si)"
 		dist_version="$(lsb_release --codename | cut -f2)"
+		if [ "$lsb_dist" = "elementary OS" ] && [ "$dist_version" = "freya" ]; then
```
Comment (jfrazelle) [inline hack/install.sh]: are there other versions, i feel like this will eventually become a switch with multiple

**Comment by thaJeztah** [comment]
Hm. I know support for elementary was added in https://github.com/docker/docker/pull/12416 

Looks like this got accidentally removed here: https://github.com/docker/docker/commit/451d3f5950b493029000c2554fe3f266ffb007e4#diff-42c8fcc34f072cecfcb04ba2928080d0L158

**Comment by jfrazelle** [comment]
Wasn't by accident :innocent:

I just don't know the version conversions, so I figured people would PR it if they wanted it badly enough

**Comment by thaJeztah** [comment]
Oh, you cheeky.... :wink:

Let's call it "regression by design" then?

**Comment by jfrazelle** [comment]
SGTM ;)

**Comment by fmoliveira** [comment]
The changes from #12416 aren't enough because it'd look for `docker-engine` package in elementary/freya repository, which doesn't exist.

To make the installer script work in elementary OS, it's needed to map each version to the corresponding Ubuntu versions.

The pull request #15000  will fix this.

**Comment by fmoliveira** [comment]
As @jfrazelle have pointed, the changes from #12416 aren't the best way to do that because the cases grow endlessly. To support all elementary OS versions we would need 3 ifs or cases.

The pull request #15001  will fix this in a better way.

**Comment by fmoliveira** [comment]
The changes from #12416 will not work because it'll look for `docker-engine` package in apt repository elementary/freya, which doesn't exist.

The changes from this pull request aren't the best way to do that because the cases grow endlessly, as @jfrazelle have previously pointed. To support all elementary OS versions we would need 3 different ifs or cases with hard coded corresponding versions.

Hard coding would be ok as a last resort, but it could use a function like the folks from NodeSource have done [here](https://github.com/nodesource/distributions/blob/master/deb/setup_0.12), to make the code easier to maintain.

The pull request #15001 will fix this in a better way by querying the upstream release directly from the operating system.

**Comment by jfrazelle** [comment]
closing in favor of https://github.com/docker/docker/pull/15001, thank you so much for contributing tho, it would be awesome if you can help us test that PR

**Pull request closed without merging**